app/components/base/button/component.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Base
+  module Button
+    class Component < ApplicationComponent
+
+      attr_reader :type, :link
+
+      BASE_CLASSES = %w[btn].freeze
+
+      PRIMARY_CLASSES = %w[btn-primary].freeze
+      SECONDARY_CLASSES = %w[btn-secondary].freeze
+      SUCCESS_CLASSES = %w[btn-success].freeze
+      DANGER_CLASSES = %w[btn-danger].freeze
+      LINK_CLASSES = %w[btn-link].freeze
+
+      BUTTON_TYPE_MAPPING = {
+        primary: PRIMARY_CLASSES,
+        secondary: SECONDARY_CLASSES,
+        success: SUCCESS_CLASSES,
+        danger: DANGER_CLASSES,
+        link: LINK_CLASSES
+      }.freeze
+
+      def initialize(type: :primary, link: '#', **options)
+        @options = options
+
+        @options[:tag] ||= :a
```

Comment:
I was wondering a few days ago how to have a clean solution that accepts both `<button>` and `<a>` tags :) 

That way is nice! 
One question though: If the tag is 'button', should we remove the default value of the `link` param?

Reply:
That makes sense. Will do that,